core/exception/backtrace_locations_spec.rb

```diff
@@ -18,14 +18,12 @@
     @backtrace.each {|l| l.should be_an_instance_of(Thread::Backtrace::Location)}
   end
 
-  it "produces a backtrace for an exception captured using $!" do
-    exception = begin
+  it "returns correct backtrace location" do
+    begin
       raise
-    rescue RuntimeError
-      $!
+    rescue RuntimeError => exception
+      exception.backtrace_locations.first.path.should =~ /backtrace_locations_spec/
     end
-
-    exception.backtrace_locations.first.path.should =~ /backtrace_locations_spec/
```

Comment:
$! not needed here.

Reply:
The example is specifically about `$!` as the description says, so it should remain.
It's fine to add new examples, but the point here is to test the backtrace of an exception captured with `$!`.